src/components/Card.js

```diff
@@ -1,41 +1,56 @@
 export default class Card{
-  #title;
+  #name;
   #link;
+  #isLiked
+  #cardId
   #cardInfo
   #cardSelector;
   #cardElement;
   #likeButton;
   #deleteButton;
   #cardImage;
   #handleImageClick
+  #confirmDeletion
+  #sendLikeInfo
+ 
+  
 
   /**
    * Code for creating new cards from a template and an object containing the text and image url
-   * @param {Object} data {link, name} two element object containing the .text and image .link with which to make the card
+   * @param {Object} item {link, name, _id} object containing the .text and image .link with which to make the card as well as the ._id of the card on the server
    * @param {Element} cardSelector the template element that gets used to build the card element
    */
-  constructor({url, title}, clickHandler, cardSelector){
-    this.#title = title;
-    this.#link = url;
-    this.#cardInfo = {title: this.#title, link: this.#link};
-    this.#handleImageClick = clickHandler.bind(this);
+  constructor(item, {handleImageClick, confirmDeletion, sendLikeInfo}, cardSelector){
+    this.#name = item.name;
+    this.#link = item.link;
+    this.#isLiked = item.isLiked;
+    this.#cardId = item._id;
+    this.#cardInfo = {title: this.#name, link: this.#link};
+    this.#handleImageClick = handleImageClick;
+    this.#confirmDeletion = confirmDeletion;
+    this.#sendLikeInfo = sendLikeInfo;
     this.#cardSelector = cardSelector;
+
+    this.#cardElement = document
+    .querySelector(this.#cardSelector)
+    .content.querySelector('.card')
+    .cloneNode(true);
+
+    this.#likeButton = this.#cardElement.querySelector('.card__button_type_like-inactive');
+    this.#deleteButton = this.#cardElement.querySelector('.card__button_type_delete');
+    this.#cardImage = this.#cardElement.querySelector('.card__image');
   }
 
   #handleLikeClick = () => {
+    this.#sendLikeInfo(this.#isLiked);
     this.#likeButton.classList.toggle('card__button_type_like-active');
```

Comment:
Needs correction: You should toggle the like button only after the successful api response.

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```